ext/standard/var.c

```diff
@@ -527,8 +527,13 @@ PHPAPI void php_var_export_ex(zval *struc, int level, smart_str *buf) /* {{{ */
 				buffer_append_spaces(buf, level - 1);
 			}
 
-			smart_str_append(buf, Z_OBJCE_P(struc)->name);
-			smart_str_appendl(buf, "::__set_state(array(\n", 21);
+			/* stdClass has no __set_state method, but can be casted to */
+			if (Z_OBJCE_P(struc) == zend_standard_class_def) {
+				smart_str_appendl(buf, "(object) array(\n", 16);
```

Comment:
The hardest two things in computer science are, of course, off-by-one errors. This was briefly `smart_str_appendl(buf, "(object) array(\n", 15);` and I was confused why there was no newline…

Reply:
`sizeof(…)-1` to the rescue?

Reply:
I wanted to do one better and use `ZEND_STRL`, but I can't because `smart_str_appendl` is a macro. I could use `smart_str_appends`, but that it uses `strlen` annoys me at some level, even though any reasonable compiler™ will do the right thing…

Reply:
Thanks, I was not aware of either! `ZEND_STRL` is … fascinating! Anyway, `smart_str_appends` might still be preferable here – after all, it's just about a potentially missing *optimization*, and I prefer clean code over any optimization, unless the latter proves to be worth the trouble. I'm fine with your current solution, though, since it doesn't make things worse.